Feature/#102 애플로그인(rest) 방식

Highjune opened the pull request:
### ✅ PR 타입 & 이슈번호
ex) 기능 추가 (#8)

### 📌 상세 내용
- 애플로그인 rest 방식 구현
- 아직 미완성(프런트에서 관련 인증코드 아직 못 받아서 테스트 못함)

### 🔥 궁금한점
- 애플에서 발급해주는 refresh Token 값으로 세션 유지하라는데, 토큰 2개(access, refresh) 는 그냥 우리가 발급 & 유지 해주면 되는거 아닌가
- signup 별도로 필요없는지

Highjune committed:
[feat] 애플로그인
- AppleAuthClient 생성

Highjune committed:
[feat] Config. 설정(.yml)에 설정값들 추가

Highjune committed:
[feat] AppleFeignClient 설정클래스 추가

Highjune committed:
[feat] Apple 설정값 들고오는 클래스 추가

Highjune committed:
[feat] Controller 에 social signup, login 추가

Highjune committed:
[feat] AppleLoginResponse 추가

Highjune committed:
[feat] 애플로그인시 identityToken 검증시 사용되는 public 키 조회 후 응답객체 생성

Highjune committed:
[feat] 애플로그인시 privateKey decode, clientSecret 키 생성 로직 추가

Highjune committed:
[feat] TokenDecoder 추가

Highjune committed:
[feat] SocialLoginRequest 객체 추가

Highjune committed:
[feat] SocialSignupRequest 객체 추가

Highjune committed:
[feat] AppleSocialTokenInfoResponse 객체 추가

Highjune committed:
[feat] AppleIdTokenPayload 객체 추가

Highjune committed:
[feat] AppleToken의 request, response 객체 추가

Highjune committed:
[feat] TokenResponse 에 isFirstValue 추가

Ryuhyis reviewed (approved):
- 애플에서 발급해주는 refresh Token 값으로 세션 유지 이게 무슨말인지 모르겠습니다. 애플에서 주는 토큰과 우리가 발급하는 jwt는 완전히 별개아닌가요

- signup 완전 필요함. 첫로그인인 경우 isFirst = True로 가니까 프론트에서 약관으로 리다이렉트할거고, 약관 동의(signup)한 경우 USER로 변경되는 로직입니다

++) PR 제목 변경 부탁합니다!

Ryuhyis commented:
> * 애플에서 발급해주는 refresh Token 값으로 세션 유지 이게 무슨말인지 모르겠습니다. 애플에서 주는 토큰과 우리가 발급하는 jwt는 완전히 별개아닌가요
> * signup 완전 필요함. 첫로그인인 경우 isFirst = True로 가니까 프론트에서 약관으로 리다이렉트할거고, 약관 동의(signup)한 경우 USER로 변경되는 로직입니다

Highjune committed:
Merge branch 'develop' into feature/#102

Highjune committed:
Merge branch 'develop' into feature/#102

# Conflicts:
#	server-api/src/main/java/com/depromeet/auth/controller/AuthController.java
#	server-api/src/main/java/com/depromeet/auth/dto/TokenResponse.java
#	server-api/src/main/java/com/depromeet/auth/service/AuthService.java

Highjune committed:
[feat] AuthController 에서 애플로그인 추가

Highjune committed:
[feat] AuthService 에 애플로그인 로직 추가
- createUser 메서드 파라미터 수정(kakao, apple 다 받을 수 있도록)
- GUEST 생성시 닉네임(게스트 + UUID 8자리) 수정
- 애플 인증서버에 토큰값 재요청 하는 로직 삭제(우리가 별도로 발급하는 토큰값 사용하므로)

Highjune committed:
[feat] 사용하지 않는 AppleProperties 의존성 삭제
